src/Symfony/Bundle/FrameworkBundle/Tests/CacheWarmer/SerializerCacheWarmerTest.php

```diff
@@ -112,8 +160,12 @@ public function testClassAutoloadExceptionWithUnrelatedException()
             }
         }, true, true);
 
-        $warmer->warmUp('foo');
+        try {
+            $warmer->warmUp(\dirname($file), \dirname($file));
+        } finally {
+            $this->assertFileDoesNotExist($file);
```

Comment:
why put this in the finally block?

Reply:
Well, it's been a while since I wrote that, but IIRC, it's because the test is purposefully making `warmUp` throws an exception, so there's no way to ensure no file gets written in that case without the finally block.

Reply:
then we should catch the expected type of exception instead and put that there